.github/workflows/ci.yml

```diff
@@ -209,3 +210,95 @@ jobs:
       - name: Mark the job as failed
         if: env.status != 'success'
         run: exit 1
+
+  #################### STAGE 4 ######################
+  ## stage4: this is for testing the new CI env
+  systemtest-dummy:
+    # only run on push, not on pull_request
+    if: always() && github.event_name == 'push'
```

Comment:
Remind me why the `always()` is necessary here. Do you want this to run even if stage1 or stage2 fails? If so, the `needs: ci-stage2` would be an issue.